src/Bicep.LangServer/Handlers/BicepDecompileForPasteCommandHandler.cs

```diff
@@ -0,0 +1,407 @@
+                    GetSuccessTelemetry(queryCanPaste, decompileId, json, pasteType, bicep: null));
+            }
+
+            // Show disclaimer and completion
+            string disclaimerMessage = "";
+            if (needsDisclaimer)
+            {
+                disclaimerMessage = $"{LangServerResources.DecompileAsPaste_AutoConvertWarning}\n{BicepDecompiler.DecompilerDisclaimerMessage}";
+                Log(output, disclaimerMessage);
+            }
+
+            // Get Bicep output from the main file (all others are currently ignored)
+            string bicepOutput = filesToSave.Single(kvp => BicepDummyUri.Equals(kvp.Key)).Value;
+
+            // Ensure ends with newline
+            bicepOutput = bicepOutput.TrimEnd() + "\n";
+
+            // Return result
+            return new ResultAndTelemetry(
+                new BicepDecompileForPasteCommandResult(decompileId, output.ToString(), pasteType, null, bicepOutput, disclaimerMessage),
+                // Don't log telemetry if we're just determining if we can paste, because this will happen a lot
+                //   (on changing between editors for instance)
+                GetSuccessTelemetry(queryCanPaste, decompileId, json, pasteType, bicepOutput));
+        }
+
+        private static void Log(StringBuilder output, string message)
+        {
+            output.AppendLine(message);
+            Trace.TraceInformation(message);
+        }
+
+        private BicepTelemetryEvent? GetSuccessTelemetry(bool queryCanPaste, string decompileId, string json, string? pasteType, string? bicep)
+        {
+            // Don't log telemetry if we're just determining if we can paste, because this will happen a lot
+            //   (on changing between editors for instance)
+            // TODO: but we don't call back for telemetry if we use the result
+            return queryCanPaste ?
+                null :
+                BicepTelemetryEvent.DecompileForPaste(decompileId, pasteType, json.Length, bicep?.Length);
+        }
+
+        /// <summary>
+        /// If the given JSON matches a pattern that we know how to paste as Bicep, convert it into a full template to be decompiled
+        /// </summary>
+        private (string? pasteType, string? fullJsonTemplate, bool needsDisclaimer) ConstructFullJsonTemplate(StringBuilder output, string json)
+        {
+            using var streamReader = new StringReader(json);
+            using var reader = new JsonTextReader(streamReader);
+            reader.SupportMultipleContent = true; // Allows for handling of lists of resources separated by commas
+
+            if (LoadValue(reader, readFirst: true) is JToken value)
+            {
+                if (value.Type == JTokenType.Object)
+                {
+                    var obj = (JObject)value;
+                    if (TryGetStringProperty(obj, "$schema") is string schema)
```

Comment:
Can we do LanguageConstants.ArmTemplateSchemaRegex.IsMatch(json) instead?

Reply:
10308054